src/Raven.Server/Documents/Sharding/Handlers/ShardedIncomingReplicationHandler.cs

```diff
@@ -189,6 +211,29 @@ protected override async Task HandleBatchAsync(TransactionOperationContext conte
                 }
 
                 _lastAcceptedChangeVector = ChangeVectorUtils.MergeVectorsDown(cvs);
+
+                if (ShardHelper.IsShardName(ConnectionInfo.SourceDatabaseName))
+                {
+                    foreach (var (shardNumber, handler) in _handlers)
```

Comment:
Let's have here a single command

Reply:
It might require creating a new state/command

Reply:
done